src/cart/test/util/cart_logtest.py

```diff
@@ -233,6 +233,48 @@ def __init__(self, log_iter):
         self.fi_triggered = False
         self.fi_location = None
 
+        # Records on number, type and frequency of logging.
+        self.log_locs = Counter()
+        self.log_mask = Counter()
+        self.log_levels = Counter()
+        self.log_count = 0
+
+    def __del__(self):
+        self.show_common_logs()
+
+    def save_log_line(self, line):
+        """Record a single line of logging"""
+        self.log_count += 1
+        loc = '{}:{}'.format(line.filename, line.lineno)
+        self.log_locs[loc] += 1
+        self.log_mask[line.mask] += 1
+        self.log_levels[line.level] += 1
+
+    def show_common_logs(self):
+        """Report to stdout the most common logging locations"""
+        print('Parsed {} lines of logs'.format(self.log_count))
+        print('Most common logging locations')
+        for (loc, count) in self.log_locs.most_common(10):
+            if count < 10:
+                break
+            print('Logging used {} times at {} ({:.1f}%)'.format(count,
+                                                                 loc,
+                                                                 100*count/self.log_count))
+        print('Most common masks')
+        for (mask, count) in self.log_mask.most_common(10):
+            if count < 10:
+                break
+            print('{}: {} ({:.1f}%)'.format(mask, count,
+                                            100*count/self.log_count))
+
+        print('Most common levels')
+        for (level, count) in self.log_levels.most_common(10):
+            if count < 10:
+                break
+            print('{}: {} ({:.1f}%)'.format(cart_logparse.LOG_NAMES[level],
+                                            count,
+                                            100*count/self.log_count))
+
```

Comment:
Would this be better as an artifact?  Just to the goal of people not having to grub through logs.

Reply:
Yes, although there's the question of if you want one per log file, one per pid in each log file or one per entire build.